account_operating_unit/models/account_invoice.py

```diff
@@ -18,6 +18,26 @@ class AccountInvoice(models.Model):
                                         states={'draft': [('readonly',
                                                            False)]})
 
+    @api.onchange('operating_unit_id')
+    def _onchange_operating_unit(self):
+        if self.operating_unit_id:
```

Comment:
`if self.operating_unit_id and (not self.journal_id or self.journal_id.operation_unit_id != self.operating_unit_id`